src/tokenize.cpp

```diff
@@ -1109,37 +1116,40 @@ static bool parse_string(tok_ctx &ctx, chunk_t &pc, size_t quote_idx, bool allow
       {
          pc.nl_count++;
          set_chunk_type(&pc, CT_STRING_MULTI);
-         escaped = false;
-         continue;
       }
-
-      if ((ch == '\r') && (ctx.peek() != '\n'))
+      else if (ch == '\r' && ctx.peek() != '\n')
       {
          pc.str.append(ctx.get());
          pc.nl_count++;
          set_chunk_type(&pc, CT_STRING_MULTI);
+      }
+
+      // if last char in prev loop was escaped the one in the current loop isn't
+      if (escaped)
+      {
          escaped = false;
          continue;
       }
 
-      if (!escaped)
+      // see if the current char is a escape char
+      if (allow_escape)
```

Comment:
Great, I think this will work properly. I messed a bit with the previous code and it still escaped `termination_character`, so it had problems with e.g. `r"\"`.